### DISCUSSION TITLE
Black screen in  ArSceneView when adding SceneView

### DISCUSSION THREAD

**Discussion by mecherguikhaoula1234** [sceneview/sceneview#349] [Q&A]
Hello,
I am trying to create an interface with 3D and AR capabilities and i switch between the visibility of ArSceneView and SceneView but when i add SceneView with ArSceneView, the camera in Ar is with black background but it detectS the real world .
NB:When i am using ArSceneView, the visibility of SceneView is Gone  and when i completely delete the SceneView, the camera in AR mode works well and display the real world.
Any idea please  ?

**Comment by ThomasGorisse** [comment]
Using both `SceneView` (3D) and `ARSceneView` (AR) simultaneously can cause conflicts because they share the Filament engine. In v4.0.0-rc.1, avoid having both active at the same time.

Instead, use a state variable to switch between them:
```kotlin
var showAR by remember { mutableStateOf(false) }

if (showAR) {
    ARSceneView { /* AR content */ }
} else {
    SceneView { /* 3D content */ }
}
```

If you need 3D content in AR, just use `ARSceneView` — it renders both the camera feed and 3D models. There's no need to layer a separate `SceneView` on top.
